### PR TITLE
feat: update tree-sitter, recognize modelines

### PR DESCRIPTION
**Pull request by ribru17**
Parses modelines based on the rules specified in `:h modeline`. Also updates the grammar to the latest Tree-sitter CLI version (available in NPM).

### CODE REVIEW

**Comment by stsewd** [comment]
Hi, thanks for the PR. I may have to think more about this, this type of modelines are specific to vim, not sure if they should be in the parser itself.

A PR to update to the latest TS cli is welcome, if you can do that in a separate PR, I'll be happy to review it.

File: binding.gyp [hunk 1]
```diff
@@ -2,18 +2,20 @@
...
+      ],
       "include_dirs": [
-        "<!(node -e \"require('nan')\")",
-        "src"
+        "src",
       ],
       "sources": [
-        "src/parser.c",
         "bindings/node/binding.cc",
-        "src/scanner.c"
```
Comment (clason) [inline binding.gyp:12]: you have to re-add the scanner manually here; `ts g` unfortunately overwrites that section.

Comment (ribru17) [reply binding.gyp:12]: thank you!

**Comment by ribru17** [comment]
@stsewd Totally, I will separate them. Feel free to merge what you want :+1:

**Pull request closed without merging**